ktor-server/ktor-server-plugins/ktor-server-sessions/common/src/io/ktor/server/sessions/SessionsBuilder.kt

```diff
@@ -343,6 +347,15 @@ internal constructor(
      * Gets a configuration used to specify additional cookie attributes for [Sessions].
      */
     public val cookie: CookieConfiguration = CookieConfiguration()
+
+    /**
+     * Ensures fields are initialized properly.
+     */
+    internal fun init() {
+        if (!this::serializer.isInitialized) {
+            serializer = defaultSessionSerializer(typeInfo)
+        }
```

Comment:
Could we use lazy property instead?
```kotlin
private var _serializer: SessionSerializer<S>? = null
private val defaultSerializer: SessionSerializer<S> by lazy { defaultSessionSerializer(typeInfo }

public var serializer: SessionSerializer<S>
    set(value) { _serializer = value }
    get() = _serializer ?: defaultSerializer
```

Reply:
Yeah, good call, fixes the temporal coupling 👍